lib/spack/spack/package.py

```diff
@@ -1343,6 +1343,10 @@ def do_install(self,
                 msg = '{0.name} is already installed in {0.prefix}'
                 tty.msg(msg.format(self))
                 rec = spack.store.db.get_record(self.spec)
+                # In case the stage directory has already been created,
+                # this ensures it's removed after we checked that the spec
+                # is installed
+                self.stage.destroy()
```

Comment:
@alalazo IMO this should only be done if `keep_stage` is `False`

Reply:
Needed to pass unit tests, unless we want to change their semantics.

Reply:
There is a unit test which sets `keep_stage` to `True` and expects the stage to be removed? If this creates a substantial amount of work I'm fine punting on that but IMO it would be good to list what goes wrong when this is done conditionally (i.e. what tests need to be updated).

Reply:
@scheibelp Sorry, I misread you first message (and thought you were suggesting deleting this line). I think the issue should be fixed now.